src/generateTypescriptClient.ts

```diff
@@ -473,16 +474,17 @@ function generateClientCode(types: ReadonlyArray<IntrospectionType>, options: Om
   return output
 }
 
-async function fetchIntrospection({ endpoint, headers }: { endpoint: string; headers?: IClientOptions['headers'] }) {
-  const introspectionCacheFileName = `gql-ts-client__introspection__${kebabCase(endpoint)}.json`
+async function fetchIntrospection({ endpoint, headers, introspectionEndpoint }: Omit< IClientOptions, 'output'>) {
+  const url = introspectionEndpoint || endpoint
```

Comment:
This definition should be done outside this function. We should keep just one `endpoint` param.